src/undo/EraseUndoAction.h

```diff
@@ -11,34 +11,33 @@
 
 #pragma once
 
+#include <set>
 #include <string>
-#include <vector>
 
+#include "PageLayerPosEntry.h"
 #include "UndoAction.h"
-#include "XournalType.h"
 
-class Layer;
-class Redrawable;
 class Stroke;
 
 class EraseUndoAction: public UndoAction {
 public:
     EraseUndoAction(const PageRef& page);
-    virtual ~EraseUndoAction();
 
 public:
-    virtual bool undo(Control* control);
-    virtual bool redo(Control* control);
+    bool undo(Control* control) override;
+    bool redo(Control* control) override;
 
     void addOriginal(Layer* layer, Stroke* element, int pos);
     void addEdited(Layer* layer, Stroke* element, int pos);
     [[maybe_unused]] void removeEdited(Stroke* element);
 
     void finalize();
 
-    virtual string getText();
+    string getText() override;
 
 private:
-    GList* edited = nullptr;
-    GList* original = nullptr;
+    std::multiset<PageLayerPosEntry<Stroke>, decltype(&PageLayerPosEntry<Stroke>::cmp)> edited{
+            PageLayerPosEntry<Stroke>::cmp};
+    std::multiset<PageLayerPosEntry<Stroke>, decltype(&PageLayerPosEntry<Stroke>::cmp)> original{
+            PageLayerPosEntry<Stroke>::cmp};
```

Comment:
This is wrong, using a three-way comparator here will violate the [Compare Requirement](https://en.cppreference.com/w/cpp/named_req/Compare).
Generally, I prefer structs to functions, because it is not required to initialize them manually.

```suggestion
    std::multiset<PageLayerPosEntry<Stroke>> edited{};
    std::multiset<PageLayerPosEntry<Stroke>> original{};
```